packages/amplication-git-utils/src/providers/bitbucket/bitbucket.service.spec.ts

```diff
@@ -280,4 +280,122 @@ describe("bitbucket.service", () => {
+              },
+            },
+            type: "user",
+            uuid: "{c3f8c1a5-185c-4fee-9bc1-bbceae764ab4}",
+            account_id: "5c0cb3e50ecb4f1b2ffaad26",
+            nickname: "amit barletz",
+          },
+        },
+        message: "Initial commit",
+        summary: {
+          type: "rendered",
+          raw: "Initial commit",
+          markup: "markdown",
+          html: "<p>Initial commit</p>",
+        },
+        links: {
+          self: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/commit/7780b0365ff60174493acc8706f104aee97308a1",
+          },
+          html: {
+            href: "https://bitbucket.org/ab-2/best-repo/commits/7780b0365ff60174493acc8706f104aee97308a1",
+          },
+          diff: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/diff/7780b0365ff60174493acc8706f104aee97308a1",
+          },
+          approve: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/commit/7780b0365ff60174493acc8706f104aee97308a1/approve",
+          },
+          comments: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/commit/7780b0365ff60174493acc8706f104aee97308a1/comments",
+          },
+          statuses: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/commit/7780b0365ff60174493acc8706f104aee97308a1/statuses",
+          },
+          patch: {
+            href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo/patch/7780b0365ff60174493acc8706f104aee97308a1",
+          },
+        },
+        parents: [],
+        rendered: {
+          message: {
+            type: "rendered",
+            raw: "Initial commit",
+            markup: "markdown",
+            html: "<p>Initial commit</p>",
+          },
+        },
+        repository: {
+          type: "repository",
+          full_name: "ab-2/best-repo",
+          links: {
+            self: {
+              href: "https://api.bitbucket.org/2.0/repositories/ab-2/best-repo",
+            },
+            html: {
+              href: "https://bitbucket.org/ab-2/best-repo",
+            },
+            avatar: {
+              href: "https://bytebucket.org/ravatar/%7B23203fef-f9de-4268-9a81-a8402af296b6%7D?ts=default",
+            },
+          },
+          name: "best-repo",
+          uuid: "{23203fef-f9de-4268-9a81-a8402af296b6}",
+        },
+      } as unknown as Commit;
+
+      const spyOnGetFirstCommitRequest = jest
+        .spyOn(requests, "getFirstCommitRequest")
+        .mockResolvedValue(mockedGetFirstCommitResponse);
+
+      const result = await service.getFirstCommitOnBranch({
+        owner: "maccheroni",
+        branchName: "master",
+        repositoryName: mockedGetFirstCommitResponse.repository.name,
+        gitGroupName:
+          mockedGetFirstCommitResponse.repository.full_name.split("/")[0],
```

Comment:
what happen if we have a repo inside a project in bitbucket?  is this expectation still valid?

Reply:
nevermind, projects are a property or a repo and they do not interfere with the repository.full_name property